ironfish-cli/src/commands/service/sync.ts

```diff
@@ -126,4 +132,27 @@ export default class Sync extends IronfishCommand {
 
     await commit()
   }
+
+  async syncTransactionGossip(api: WebApi, maxUpload: number): Promise<void> {
+    const client = await this.sdk.connectRpc()
+
+    const response = client.event.onTransactionGossipStream({})
+
+    const buffer = new Array<Transaction>()
+
+    async function commit(): Promise<void> {
+      await api.transactions(buffer)
+      buffer.length = 0
```

Comment:
does this actually clear the buffer? Never seen this before

Reply:
yeah this looks a bit odd 🤔 

copied from the existing syncer code, but i'll take a closer look

Reply:
This will clear the array without replacing it: https://stackoverflow.com/a/1232046